src/components/Dropdown.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import Select  from 'react-select';
+import { stat } from 'fs';
+import { fetchStations } from '../helpers/ApiCallHelper';
+import { responseMapper } from '../mappers/ResponseMapper';
+import {  Station } from '../models/Station';
+import Button from './Button';
+
+type DropdownProps = {
+    onChange: (option: React.SetStateAction<Station | undefined> ) => void;
+    stationsList: Station[];
+    selectedOption: Station | undefined;
+}
+
+const DropdownMenu: React.FC<DropdownProps>  = ({ onChange, selectedOption, stationsList }) => {
+    const handleSelectChange = (option?: Station | null | undefined) => {
```

Comment:
Do we need `?` and `null` and `undefined`?

Reply:
When remuving it a " Argument of type 'Station | undefined' is not assignable to parameter of type 'Station'" error comes up on line 24 in the useEffect

Reply:
I am able to get rid of ?